src/cmdlib.sh

```diff
@@ -601,9 +601,14 @@ runcompose_extensions() {
 # the cache disk. `runvm_with_cache_snapshot on` will set snapshotting to on.
 runvm_with_cache_snapshot() {
     local snapshot=$1; shift
+    local cache_size="16G"
+    if [  "$1" != "--" ]; then
+        cache_size=${1}; shift
+    fi
```

Comment:
I don't think this will do the right thing. It's supported to pass QEMU args before the `--`.

Since the cache size is set once on creation and then has no effect on future invocations of `runvm`, it might be better to model this differently. How about just an env var (e.g. `RUNVM_CACHE_SIZE`)? `cmd-supermin-run` can set it and here we do `${RUNVM_CACHE_SIZE:-16G}` when passing to `qemu-img create`.

Reply:
Yes, much better, thanks for the suggestion!